pkg/runtime/runtime.go

```diff
@@ -18,10 +19,13 @@ var (
 func updateState(ctx context.Context, syncr sync.ISync) error {
 	msg, err := syncr.Fetch(ctx)
 	if err != nil {
-		return err
+		return fmt.Errorf("fetch: %w", err)
 	}
 	mu.Lock()
-	_ = ev.SetState(msg)
+	err = ev.SetState(msg)
+	if err != nil {
+		return fmt.Errorf("set state: %w", err)
+	}
 	mu.Unlock()
```

Comment:
This mutex will never unlock, can we defer it before the set state?

Reply:
@skyerus ^

Reply:
Nice spot, done. I'm surprised this wasn't picked up by the linter